torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -24,308 +24,376 @@ namespace jit {
+         listAppend,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::reverse.t(t[](a!) self) -> ()",
+         listReverse,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::extend.t(t[](a!) self, t[] other) -> ()",
+         listExtend,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::copy.t(t[](a) self) -> t[]",
+         listCopy,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::_set_item.t(t [](a!) l, int idx, t(b -> *) el) -> t[](a!)",
+         listSetItem,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::clear.t(t[](a!) self) -> ()",
+         listClear,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::Delete.t(t[](a!) self, int idx) -> ()",
+         listDelete,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::insert.t(t[](a!) self, int idx, t(b -> *) el) -> ()",
+         listInsert,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::pop.t(t[](a!) self, int idx=-1) -> t(*)",
+         listPop,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::add.t(t[] a, t[] b) -> t[]",
+         listAdd,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::add_.t(t[](a!) self, t[] b) -> t[]",
+         listInplaceAdd,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::slice.t(t[] l, int start, int end=9223372036854775807, int step=1) -> t[]",
+         listSlice,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::list.t(t[] l) -> t[]",
+         listList,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::mul.left_t(t[] l, int n) -> t[]",
+         listMulIntLeft,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::mul.right_(int n, t[] l) -> t[]",
+         listMulIntRight,
+         aliasAnalysisFromSchema()),
+     Operator(
+         "aten::mul_.t(t[](a!) l, int n) -> t[](a!)",
+         listMulIntLeftInPlace,
+         aliasAnalysisFromSchema()),
+     Operator("aten::len.t(t[] a) -> int", listLen, aliasAnalysisFromSchema()),
+     DEFINE_COMPARISON_OP(aten::eq, a == b),
+     DEFINE_COMPARISON_OP(aten::ne, a != b),
+     DEFINE_COMPARISON_OP(aten::lt, a < b),
+     DEFINE_COMPARISON_OP(aten::gt, a > b),
+     DEFINE_COMPARISON_OP(aten::le, a <= b),
+     DEFINE_COMPARISON_OP(aten::ge, a >= b),
+
+     //
+     // create a clone of these declarations with a _hacked_twin overload name
+     // and nullability scrubbed from TensorList arg types
+     // TOOD find out why this exists and how to do it without the hack
+     //
+     Operator(
+         "aten::index.Tensor_hacked_twin(Tensor self, Tensor[] indices) -> Tensor",
+         [](Stack& stack) {
+           auto indices = pop(stack).toTensorVector();
+           auto self = pop(stack).toTensor();
+           auto result = at::index(std::move(self), std::move(indices));
```

Comment:
hm I'll remove it for now but we should probably disable that warning, I think call sites ought to be allowed to use std::move even if the target function takes by reference, because it's not visible at the call site what the target function does and checking code correctness should be as local as possible.

Reply:
I've mentioned this to you previously in code review: I don't think you should specify `std::move` unless it semantically makes sense to actually move into the function in question. I don't think you can just rely on people to write code where they won't continue to use the value after moving it, since it won't actually be moved out of.